src/backend/core/tests/test_models_invitations.py

```diff
@@ -73,6 +73,35 @@ def test_models_invitations__is_expired(settings):
     assert expired_invitation.is_expired is True
 
 
+def test_models_invitation__new_user__convert_invitation_to_access():
+    """
+    Upon creating a new identity, invitations linked to that email
+    should be converted to accesses and then deleted.
+    """
+    invitation = factories.InvitationFactory()
+    new_identity = factories.IdentityFactory(is_main=True, email=invitation.email)
+
+    assert models.Invitation.objects.exists() is False
+    assert models.TeamAccess.objects.exists() is True
+    assert invitation.team.get_abilities(new_identity.user)["get"] is True
+
+
+def test_models_invitation__new_user__expired_invitation(settings):
+    """
+    Upon creating a new identity, expired invitations
+    should not be converted into accesses.
+    """
+    invitation = factories.InvitationFactory()
+    settings.INVITATION_VALIDITY_DURATION = 1
+    time.sleep(1)
```

Comment:
cool, using freeze gun could help us testing the default validity duration

Reply:
I still think it's a bit overkill for unit tests such as these.